src/main/java/com/google/research/bleth/servlets/EnqueueSimulationServlet.java

```diff
@@ -0,0 +1,60 @@
+package com.google.research.bleth.servlets;
+
+import com.google.cloud.tasks.v2.AppEngineHttpRequest;
+import com.google.cloud.tasks.v2.CloudTasksClient;
+import com.google.cloud.tasks.v2.HttpMethod;
+import com.google.cloud.tasks.v2.QueueName;
+import com.google.cloud.tasks.v2.Task;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A servlet used for enqueuing a task targeted at endpoint '/new-simulation',
+ * in order to create and run a new simulation.
+ */
+@WebServlet("/enqueue-simulation")
+public class EnqueueSimulationServlet extends HttpServlet {
+
+    static final String projectId = "bleth-2020";
```

Comment:
You can get this from `com.google.appengine.api.utils.SystemProperty.applicationId()`, AFAIK.

Reply:
The following exception is thrown: `com.google.api.gax.rpc.UnavailableException: io.grpc.StatusRuntimeException: UNAVAILABLE: Credentials failed to obtain metadata`
I have tried to figure it out and it seems like a possible cause might be incorrect timezones in containerized / virtualized environments (I might be wrong though). 
Isn't it safer to just hard code it, or define it somewhere inside the project itself?

Reply:
I suspect you can only call this from within a running servlet, not from initialization code (i.e. it cannot be a constant).